app/src/main/java/com/example/sharewhatyoucanproject/DonorActivity.kt

```diff
@@ -0,0 +1,166 @@
+        storageReference = FirebaseStorage.getInstance().reference
+
+        // Assign FirebaseDatabase instance with root database name.
+        databaseReference = FirebaseDatabase.getInstance().getReference(Database_Path)
+
+        //Assign ID'S to button.
+        ChooseButton = findViewById<View>(R.id.btnChoose) as Button
+        UploadButton = findViewById<View>(R.id.btnUpload) as Button
+
+
+        // Assign ID's to EditText.
+        ImageName = findViewById<View>(R.id.foodname) as EditText
+        ImageDescription = findViewById<View>(R.id.fooddesc) as EditText
+
+        // Assign ID'S to image view.
+        SelectImage = findViewById<View>(R.id.imageView) as ImageView
+
+        // Assigning Id to ProgressDialog.
+        progressDialog = ProgressDialog(this@DonorActivity)
+
+        // Adding click listener to Choose image button.
+        ChooseButton!!.setOnClickListener { // Creating intent.
+            val intent = Intent()
+
+            // Setting intent type as image to select image from phone storage.
+            intent.type = "image/*"
+            intent.action = Intent.ACTION_GET_CONTENT
+            startActivityForResult(
+                Intent.createChooser(intent, "Please Select Image"),
+                Image_Request_Code
+            )
+        }
+
+
+        // Adding click listener to Upload image button.
+        UploadButton!!.setOnClickListener { // Calling method to upload selected image on Firebase storage.
+            UploadImageFileToFirebaseStorage()
+        }
+    }
+
+
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == Image_Request_Code && resultCode == RESULT_OK && data != null && data.data != null) {
+            FilePathUri = data.data
+            try {
+
+                // Getting selected image into Bitmap.
+                val bitmap = MediaStore.Images.Media.getBitmap(contentResolver, FilePathUri)
+
+                // Setting up bitmap selected image into ImageView.
+                SelectImage!!.setImageBitmap(bitmap)
+
+                // After selecting image change choose button above text.
+                ChooseButton!!.text = "Image Selected"
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+        }
+    }
+
+    // Creating Method to get the selected image file Extension from File Path URI.
+    fun GetFileExtension(uri: Uri?): String? {
+        val contentResolver = contentResolver
+        val mimeTypeMap = MimeTypeMap.getSingleton()
+
+        // Returning the file Extension.
+        return mimeTypeMap.getExtensionFromMimeType(contentResolver.getType(uri!!))
+    }
+
+    // Creating UploadImageFileToFirebaseStorage method to upload image on storage.
+    fun UploadImageFileToFirebaseStorage() {
+
+       //here uploading image to firebase part will be implemented
```

Comment:
Maybe instead of 
```
//here uploading image to firebase part will be implemented
```
use something like
```
// TODO: Implement uploading of the image to the backend, <jira ticket>
```
Sometimes folks wire up their IDE to throw a warning when they see a TODO, so you don't forget to do it later.

Replace `<jira ticket>` with the jira ticket number or link and then reference this bit of code in your ticket, so you can find it later when you start working on the next ticket.

But if it's part of this ticket and you'll be implementing the image upload in a follow-up PR, then never mind:).